pkg/apis/v1alpha1/serviceexport.go

```diff
@@ -88,3 +92,133 @@ type ServiceExportList struct {
+	ServiceExportConditionValid ServiceExportConditionType = "Valid"
+
+	// ServiceExportReasonValid is used with the "Valid" condition when the
+	// condition is True.
+	ServiceExportReasonValid ServiceExportConditionReason = "Valid"
+
+	// ServiceExportReasonNoService is used with the "Valid" condition when
+	// the associated Service does not exist.
+	ServiceExportReasonNoService ServiceExportConditionReason = "NoService"
+
+	// ServiceExportReasonInvalidServiceType is used with the "Valid"
+	// condition when the associated Service has an invalid type
+	// (per the KEP at least the ExternalName type).
+	ServiceExportReasonInvalidServiceType ServiceExportConditionReason = "InvalidServiceType"
+)
+
+const (
+	// ServiceExportConditionExported is true when the service is exported to some
+	// control plane or data plane.
+	//
+	//
+	// Possible reasons for this condition to be true are:
+	//
+	// * "Exported"
+	//
+	// Possible reasons for this condition to be False are:
+	//
+	// * "Invalid"
+	// * "Pending"
+	//
+	// Possible reasons for this condition to be Unknown are:
+	//
+	// * "Pending"
+	//
+	// Controllers may raise this condition with other reasons,
+	// but should prefer to use the reasons listed above to improve
+	// interoperability.
+	ServiceExportConditionExported ServiceExportConditionType = "Exported"
+
+	// ServiceExportReasonExported is used with the "Exported" condition
+	// when the condition is True.
+	ServiceExportReasonExported ServiceExportConditionReason = "Exported"
+
+	// ServiceExportReasonInvalid is used with the "Exported" condition
+	// when the exported service is invalid and will not be exported.
+	ServiceExportReasonInvalid ServiceExportConditionReason = "Invalid"
+
+	// ServiceExportReasonPending is used with the "Exported" condition
+	// when the service is going to be exported.
+	ServiceExportReasonPending ServiceExportConditionReason = "Pending"
+)
+
+const (
+	// ServiceExportConditionConflict indicates that the controller was unable
+	// to resolve conflict for a ServiceExport. This condition must be at
+	// least raised on the conflicting ServiceExport and is recommended to
+	// be raised on all on all the constituent `ServiceExport`s if feasible.
+	//
+	//
+	// Possible reasons for this condition to be true are:
+	//
+	// * "PortsConflict"
+	// * "TypeConflict"
+	// * "SessionAffinityConflict"
+	// * "SessionAffinityConfigConflict"
+	// * "MultipleConflicts"
+	//
+	// Possible reasons for this condition to be False are:
+	//
+	// * "NoConflicts"
+	//
+	// Controllers may raise this condition with other reasons,
+	// but should prefer to use the reasons listed above to improve
+	// interoperability.
+	ServiceExportConditionConflict ServiceExportConditionType = "Conflict"
+
+	// ServiceExportReasonPortsConflict is used with the "Conflict" condition
+	// when the exported service have a conflict related to ports.
+	ServiceExportReasonPortsConflict ServiceExportConditionReason = "PortsConflict"
```

Comment:
Adding the specific port conflict language here from KEP-1645 would be helpful in determining when this condition should be set.

```suggestion
	// ServiceExportReasonPortConflict is used with the "Conflicted" condition
	// when a derived service would have a conflicting port configuration. This
	// includes when ports on exported services have duplicated names (including
	// unnamed/empty name) for different protocol and port number pairs. The
	// conflict resolution policy will determine which of the duplicated ports are
	// used by the ServiceImport.
	ServiceExportReasonPortConflict ServiceExportConditionReason = "PortConflict"
```

Reply:
Reading the KEP more carefully it makes sense to me that the `union` language and conflict resolution policy was intended to prevent disrupted traffic in the case of non-matching exported services, but we likely want some conformance testing around if/how unioned ports from non-matching backends should be routed (should `foo.clusterset.local:80` potentially fail if port 80 is only exposed on some subset of backends and a backend without that port open is selected, or should an implementation ensure requests to that port are always routed to eligible backends?)

Defining that behavior more precisely shouldn't block this change though.